### ISSUE TITLE
Support more lock states

### ISSUE THREAD

**Issue by ZzetT** [kvj/hass_nuki_ng#35]
When I open or close the door, I can see that the state of my Nuki is either "locked" or "unlocked". However, according to https://developer.nuki.io/page/nuki-bridge-http-api-1-13/4/#heading--lock-states also other states like "unlatched" are support, but I don't see this in HA.

Could you please add this?

**Comment by kvj** [comment]
It's already supported, you just need to use the right sensor. `binary_sensor.nuki_<...>_locked` has only two states: locked and unlocked, but `sensor.nuki_<...>_state` has more (all supported) states.

**Comment by ZzetT** [comment]
Hi @kvj ,
my lock is called "zuhause" and this is the log of the sensor `sensor.nuki_zuhause_state`:

![hascreenshot](https://user-images.githubusercontent.com/6074452/150697054-e825c539-ddf5-4b54-bb7d-c3cab484194d.png)

Here, I can only see the states "locked" and "unlocked".

**Comment by alexdelprete** [comment]
@ZzetT @Inferi0r: the unlatched state is an INTERMEDIATE state, the lock is unlatched and then unlocked (final state), the callback by the bridge to HA is triggered only on final states, no intermdiate states can be reflected by HA sensors.

This is the way the bridge firmware works, as [described](https://developer.nuki.io/t/http-api-lockstate-and-the-unlatched-state/4237/7?u=alexdelprete) by Nuki devs in their forum:

![image](https://user-images.githubusercontent.com/7027842/152084597-537564a4-b452-4a7c-b651-5899c0ef2e4e.png)

![image](https://user-images.githubusercontent.com/7027842/152084586-214b7bfe-7c8e-4a35-8d01-52871ae9b1e5.png)

**Comment by kvj** [comment]
Yeah, I also have an integration relying on the `unlatched` state change, and it's not super-reliable.

**Comment by ZzetT** [comment]
Thanks for the clarification.

**Comment by alexdelprete** [comment]
> Yeah, I also have an integration relying on the `unlatched` state change, and it's not super-reliable.

I'm very upset about the bridge and its implementation, they didn't do a very good job. When I spoke to the product manager I told him about this...but...they look at the big numbers, and we're less than 0.5% of their average user unfortunately. 99.5% of users don't have a clue of what we're talking about...they simply use the mobile with the web interface, etc.

**Comment by alexdelprete** [comment]
> Thanks for the clarification.

my pleasure. I spent days building my Nuki Card, so I spent days on their forum trying to overcome issues. I understand how you feel, but unfortunately the bridge is a badly designed solution, and I discussed with them on their forum and also in video conference with product management. But I don't expect anything good coming from those conversations, they are focused on the vast majority of users....and we're not part of them with our requirements.

**Comment by alexdelprete** [comment]
@kvj you can close this one.
